dask/dataframe/io/parquet/core.py

```diff
@@ -95,6 +95,64 @@ def __call__(self, part):
         )
 
 
+ParquetFunctionWrapper = ReadParquetFunctionWrapper
+
+
+class ToParquetFunctionWrapper:
+    """
+    Parquet Function-Wrapper Class
+
+    Writes a DataFrame partition into a distinct
+    parquet file. When called, the function also
+    requires a ``partition_info`` kwarg.
+    """
+
+    def __init__(
+        self,
+        engine,
+        path,
+        fs,
+        partition_on,
+        write_metadata_file,
+        i_offset,
+        name_function,
+        kwargs_pass,
+    ):
+        self.engine = engine
+        self.path = path
+        self.fs = fs
+        self.partition_on = partition_on
+        self.write_metadata_file = write_metadata_file
+        self.i_offset = i_offset
+        self.name_function = name_function
+        self.kwargs_pass = kwargs_pass
+
+    def __call__(self, df, partition_info=None):
+
+        # Get partition index
+        part_i = (partition_info or {}).get("number")
+        filename = (
+            f"part.{part_i + self.i_offset}.parquet"
+            if self.name_function is None
+            else self.name_function(part_i + self.i_offset)
+        )
+
+        # Write out data
+        return self.engine.write_partition(
+            df,
+            self.path,
+            self.fs,
+            filename,
+            self.partition_on,
+            self.write_metadata_file,
+            **(
+                toolz.merge(self.kwargs_pass, {"head": True})
```

Comment:
```suggestion
               dict(self.kwargs_pass, head=True)
```
nit: Is `toolz.merge` faster? Otherwise, I lean slightly towards pure Python